### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lefteris-L**
[DEMO LINK](https://lefteris-l.github.io/react_clock-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -8,32 +10,45 @@ function getRandomName(): string {
...
+    const timerId = window.setInterval(
+      () => setClockName(getRandomName()),
+      3300,
+    );
+
+    return () => {
+      document.removeEventListener('click', onClick);
+      document.removeEventListener('contextmenu', onRightClick);
+      window.clearInterval(timerId);
+    };
```
Comment (kenvastolord) [inline src/App.tsx:42]: this is an excellent job!, you remembered to removing the eventListener , Keep doing this!

**Review by kenvastolord — COMMENTED** [review]
From my point of view, You did an excellent job, You made it look easy!

**Comment by lefteris-L** [comment]
> From my point of view, You did an excellent job, You made it look easy!

Very kind. Thanks man 😃

File: src/App.tsx [hunk 2]
```diff
@@ -8,32 +10,45 @@ function getRandomName(): string {
...
+
+  const onRightClick = (event: MouseEvent) => {
+    event.preventDefault();
+    setHasClock(false);
+  };
+
+  const onClick = () => setHasClock(true);
+
+  useEffect(() => {
+    if (hasClock && clockName !== 'Clock-0') {
```
Comment (naziamoff) [inline src/App.tsx:24]: `Clock-0` looks hardcoded here. What if we will change the default name later? Better to create variable for that

**Review by naziamoff — APPROVED** [review]
Well done. Just one suggestion

**Pull request closed without merging**